controllers/idle_handlers.go

```diff
@@ -18,16 +18,57 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"os"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// removeDeployment removes the deployment with the specified name
+// in case there is no deployment with the specified name returns false
+func (r *ImageBasedUpgradeReconciler) removeDeployment(osName string) (bool, error) {
+	status, err := r.RpmOSTreeClient.QueryStatus()
+	if err != nil {
+		return false, err
+	}
+	for index, deployment := range status.Deployments {
+		if deployment.OSName == osName {
+			ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy %d", index)
+			if err := utils.ExecuteChrootCmd(utils.Host, ostreeUndeployCommand); err != nil {
+				return true, err
+			}
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
 //nolint:unparam
 func (r *ImageBasedUpgradeReconciler) handleAbort(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	OSName := getOSName(ibu.Spec.SeedImageRef.Version)
```

Comment:
This function should also be renamed to getStateroot. Currently, this function seems geared toward just the seed image version? We'd also need to clean up the original stateroot, which could be rhcos or rhcos_$version, after the upgrade is completed. I assume the handleAbort is only called from the A-side?